spec.html

```diff
@@ -8340,6 +8356,28 @@ <h1>ECMAScript Function Objects</h1>
             *true* if this is a strict function, *false* if this is a non-strict function.
           </td>
         </tr>
+        <tr>
+          <td>
+            [[OmittedFromStackTraces]]
+          </td>
+          <td>
+            Boolean
+          </td>
+          <td>
+            When *true*, this function must not have a corresponding entry in stack traces.
+          </td>
+        </tr>
+        <tr>
+          <td>
+            [[ImplementationHidden]]
```

Comment:
Maybe something more like the slot for toString, `[[HasSourceTextAvailable]]`?

I’d also expect one slot not to apply the other; iow, I’d expect both to be required to hide both.

I’ll leave a comment on the proposal issue about the bikeshed; i don’t think we’ve landed on the best pair of wordings quite yet :-)

Reply:
I'm sorry @ljharb but I don't understand any part of that comment.

Reply:
Sure, I'll rephrase: your current implementation allows these two slots to function independently. This means that, although the proposal only has three states: "true and true", "true and false", or "false and false", four states are possible.

Perhaps an enum, rather than two booleans, is a better choice; but if it's going to be two booleans, than I'd prefer to see the implementation explicitly prevent "false and true" from having any effect (or better, an assertion that this is not the case).

Reply:
The proposal has two directives, each of which may be used independently. Four states is the exact number needed to represent these combinations. What is "false and true" and why is it not meaningful?

Reply:
I've renamed `ImplementationHidden` to `HasSourceTextAvailable` and renamed `OmittedFromStackTraces` to `PresentInStackTraces`.

Reply:
maybe I’m confused; i thought there were two pragmas in the proposal; is the intention to require both to be present, or does one subsume the other?

Reply:
There are two directives and two features. The `hide source` directive enables one feature, and the `sensitive` directive enables both features. So you could say that `sensitive` subsumes `hide source`.

Reply:
Right - so there's only three states:
1. no features
1. one feature (hide source)
1. both features (hide source, hide from stacks)

Two booleans represents four states, and I'd prefer to see either three (an enum) or an explicit assertion that makes it impossible for "hide from stacks but show source" to be a thing.

Reply:
Why are hide source and hide from stack coupled at all?  I have actually built a debugger (for another language) where the frames of  certain implementation generated functions were (by default) hidden from stack traces.  This was done to make a simpler presentation that avoided burdening the user will seeing those implementation details.  But hiding the stack frame had nothing to do with whether or not source code of such functions was accessible.  That should be an orthogonal option leaving open the possible that a function's source can be accessed even if it has hidden stack frames.

I think you should have two boolean valued slots associated with each function: [[HiddenSource]] and [[HiddenStack]] generated four states, each of which is meaningful.  Also isHiddenSpirce and isHiddenStack abstract operations. 

BTW,  the setting and access of these should exactly parallel the pattern used for [[Strict]]/isStrict.

Reply:
If the current proposal allows users to have that use case, then I agree that two booleans is the right way to implement it; however, it currently does not. When/if such a use case is supported, we can refactor the spec to use two booleans at that time.

Reply:
That sounds silly.  Fix the proposal to do the rational thing.

Reply:
@allenwb I'd appreciate your feedback on the proposal repo. We arrived at this design through careful consideration of the use cases we care about. See https://github.com/tc39/proposal-function-implementation-hiding/issues/35.

Reply:
@ljharb I've added an assertion, please check it out.

Reply:
Thanks, that resolves my concern!